lib/crypt.c

```diff
@@ -334,8 +353,9 @@ crypt_checksalt (const char *setting)
 {
   int retval = CRYPT_SALT_INVALID;
 
-  if (!setting ||          /* NULL string */
-      setting[0] == '\0')  /* empty passphrase, or descrypt without salt */
+  if (!setting ||            /* NULL string */
+      setting[0] == '\0' ||  /* empty passphrase, or descrypt without salt */
+      check_badsalt_chars (setting))  /* bad salt chars */
     goto end;
```

Comment:
FWIW, I don't understand what "descrypt without salt" means in the comment here (unchanged in this PR). I think there's no such thing as "descrypt without salt".

Reply:
Fixed in rebased commit.